### PR TITLE
Make close button on modals accessible.

### LINKED ISSUE #5520: (WCAG 2.1) Lack of Accessible Label for Close Control
OBSERVED RESULTS:

- The Close control (dismiss button) on the "Create an Account or Sign In" modal dialog lacks an accessible label. The control is currently implemented as an SVG image, making it unclear to screen reader users.
- Screenshot

EXPECTED RESULTS:
 
- The Close control on the modal dialog should have a meaningful and accessible label that conveys its purpose to all users, including screen reader users.
- Implement the Close control using an HTML <button> element with an appropriate aria-label attribute. Ensure that the aria-label attribute provides a concise description of the action, such as "Close" or "Dismiss."

STEPS TO REPRODUCE:

- Enable screen reader.
- Navigate to [URL](https://demo.civiform.dev/)
- Navigate to and activate the ‘Apply’ for the Comprehensive sample program.
- Navigate to and activate “What is your address?”.
- Fillup the form fields and complete the application.
- On the confirmation page navigate to and activate “Apply to Another Program" using the tab key.
- Navigate to the “close” button and observe that the button is not keyboard accessible and the screen reader doesn't announce it as a close/dismiss button.

ENVIRONMENT DETAILS:

- Device(s) (brand, OS version, test key or release key): Windows 10 Enterprise (64-bit),MacOS Ventura Version 13.4
- Screen reader name and version: JAWS 2022.2211.7, NVDA 2022.3, VoiceOver
- Browser name and version: Chrome Version 113.0.5672.93

### LINKED ISSUE #5521: (WCAG 2.1) Close control is not keyboard accessible
OBSERVED RESULTS:

- The Close control on a “Create an account or sign in” modal dialog is not keyboard accessible and cannot be operated using keyboard navigation.
- Screenshot

EXPECTED RESULTS:

- The Close control on a “Create an account or sign in” modal dialog should be fully keyboard accessible, allowing users to close the modal using the keyboard.
- Use a role button and or use <button> to implement the close button.
- Provide valid label using aria-label to the close button.

STEPS TO REPRODUCE:

- Navigate to [URL](https://demo.civiform.dev/)
- Navigate to and activate the ‘Apply’ for the Comprehensive sample program.
- Navigate to and activate “What is your address?”.
- Fillup the form fields and complete the application.
- On the confirmation page navigate to and activate “Apply to Another Program" using the tab key.
- Navigate to the “close” button using TAB key and observe that the control is not keyboard accessible.

ENVIRONMENT DETAILS:

- Device(s) (brand, OS version, test key or release key): Windows 10 Enterprise (64-bit),MacOS Ventura Version 13.4
- Screen reader name and version: JAWS 2022.2211.7, NVDA 2022.3, VoiceOver
- Browser name and version: Chrome Version 113.0.5672.93

### PR DESCRIPTION
**Pull request by caitlinshk**
### Description

A few pieces in this PR:
1) Updates the close button icon to be inside a `<button>` with a valid aria-label.
2) Since the aria-label needs to be translated for applicant-facing modals, updates the `Modal.Builder` to take in (1) A `Location`, which can be `ADMIN_FACING`, `APPLICANT_FACING`, or `DEBUG`; and (2) An optional `Messages` object that has the translated messages. If the `Location` is `APPLICANT_FACING`, then the modal is *required* to set the `Messages` object so that the close button aria label can be correctly translated.
3) Moves the close button to be at the end of the DOM instead of in the header, per https://designsystem.digital.gov/components/modal/.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings) -- **Will wait until translations for "Close" have merged before merging this PR**
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order -- **Specifically verified the close button is last in the tab order within the modal.**

### Instructions for manual testing

1) Start a new session on CiviForm and start an application. Verify that the close button on the "Create an account" modal can be tabbed to, has a description read out by screen reader, and that pressing enter activates the button and closes the dialog. 
2) Verify the close button description changes based on the selected language.
3) As an admin, try adding questions to a program. Verify the items listed in step 1 but with the close button on the question bank. 

### Issue(s) this completes

Fixes #5520; Fixes #5521

### CODE REVIEW

**Review by avaleske — COMMENTED** [review]
Thanks for updating the styles for this, and linking out to USWDS docs, it's super helpful.

I wonder if we need to add the complexity of the `TranslationStrategy` pattern. Instead, would either of these work?
- a) Just let the admin-facing aria label be translated. (I don't know what happens if you have some translated text on the admin side, since there's not a language picker. It might just default to english anyway)
- b) Replace this with a `isApplicantFacing()` method on the modal builder. If true, the modal pulls the close button string from the messages file. If not, it just defaults to "Close". This is essentially what we're doing with the translation strategy, right?

**Comment by caitlinshk** [comment]
Talked offline. Updated to have the Modal API be more like "where is this modal displayed?" instead of "what translation strategy to use?". If the modal is applicant-facing, the modal must also set what `Messages` object to use for translation.

File: server/app/views/BaseHtmlView.java [hunk 1]
```diff
@@ -50,6 +50,17 @@ public static DivTag fieldErrors(
...
   }
 
+  /**
+   * Creates a button that doesn't display any text but includes an aria label, which is needed for
+   * accessibility.
+   *
+   * @param ariaLabel a label that will be used by screenreaders and other accessibility services to
+   *     describe the button's purpose.
+   */
+  public static ButtonTag noTextButton(String ariaLabel) {
```
Comment (avaleske) [inline server/app/views/BaseHtmlView.java:60]: **suggestion**: would `iconButton` but a more descriptive name for this?

File: server/test/views/components/ModalTest.java [hunk 2]
```diff
@@ -68,4 +79,78 @@ public void buildsModal_withRepeatOpenBehavior_onlyShowOnceWithBypassUrl() {
...
+    assertThatThrownBy(modalBuilder::build).isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void buildsModal_applicantFacingHasMessages_usesCloseFromMessage() {
+    Lang lang = Lang.forCode("es-US");
+
+    // Note: There's a bit of an API mismatch in Play, where [stubMessagesApi] expects
+    // classes in the play.i18n package but [stubLangs] expects classes in the
+    // play.api.i18n package, which is why the same Lang is created twice.
```
Comment (avaleske) [inline server/test/views/components/ModalTest.java:130]: That's unfortunate! I'm was also surprised to learn that we've never tried to mock the messages themselves so I'm glad you figured this out.

**Review by avaleske — APPROVED** [review]
One naming question but overall lgtm!

(Also can we update the description with the new design?)

File: server/app/views/BaseHtmlView.java (hunk 1 above)
Comment (caitlinshk) [reply server/app/views/BaseHtmlView.java:60]: I like `iconOnlyButton`, done.

File: server/test/views/components/ModalTest.java (hunk 2 above)
Comment (caitlinshk) [reply server/test/views/components/ModalTest.java:130]: Yeah, it's a bit clunky but does seem to work. If we need to do more of it elsewhere we  can make a utility method at that point.

**Pull request merged**